File: sdks/sdkwork-rtc-sdk/sdkwork-rtc-sdk-go/rtcstandard/capability_catalog.go
package rtcstandard

type RtcCapabilityCatalogEntry struct {
    CapabilityKey string
    Category      string
    Surface       string
}

type RtcCapabilityCatalog struct{}

var RTC_CAPABILITY_CATALOG = []RtcCapabilityCatalogEntry{
    {CapabilityKey: "session", Category: "required-baseline", Surface: "cross-surface"},
    {CapabilityKey: "credential", Category: "required-baseline", Surface: "control-plane"},
    {CapabilityKey: "provider.webhook", Category: "required-baseline", Surface: "control-plane"},
    {CapabilityKey: "provider.event-normalization", Category: "required-baseline", Surface: "control-plane"},
    {CapabilityKey: "health", Category: "required-baseline", Surface: "control-plane"},
    {CapabilityKey: "media.audio", Category: "required-baseline", Surface: "runtime-bridge"},
    {CapabilityKey: "media.video", Category: "required-baseline", Surface: "runtime-bridge"},
    {CapabilityKey: "live.broadcast", Category: "required-baseline", Surface: "cross-surface"},
    {CapabilityKey: "live.audience", Category: "required-baseline", Surface: "cross-surface"},
    {CapabilityKey: "screen-share", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "recording", Category: "optional-advanced", Surface: "control-plane"},
    {CapabilityKey: "artifact", Category: "optional-advanced", Surface: "control-plane"},
    {CapabilityKey: "cloud-mix", Category: "optional-advanced", Surface: "control-plane"},
    {CapabilityKey: "cdn-relay", Category: "optional-advanced", Surface: "control-plane"},
    {CapabilityKey: "data-channel", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "transcription", Category: "optional-advanced", Surface: "control-plane"},
    {CapabilityKey: "beauty", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "spatial-audio", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "e2ee", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "provider.active-query", Category: "optional-advanced", Surface: "control-plane"},
}

func GetRtcCapabilityCatalog() []RtcCapabilityCatalogEntry {
    return append([]RtcCapabilityCatalogEntry(nil), RTC_CAPABILITY_CATALOG...)
}

func GetRtcCapabilityDescriptor(capabilityKey string) *RtcCapabilityCatalogEntry {
    for index := range RTC_CAPABILITY_CATALOG {
        if RTC_CAPABILITY_CATALOG[index].CapabilityKey == capabilityKey {
            return &RTC_CAPABILITY_CATALOG[index]
        }
    }

    return nil
}

File: sdks/sdkwork-rtc-sdk/sdkwork-rtc-sdk-python/sdkwork_rtc_sdk/capability_catalog.py
from dataclasses import dataclass
from typing import Optional


@dataclass(frozen=True)
class RtcCapabilityCatalogEntry:
    capabilityKey: str
    category: str
    surface: str


class RtcCapabilityCatalog:
    entries = [
        RtcCapabilityCatalogEntry("session", "required-baseline", "cross-surface"),
        RtcCapabilityCatalogEntry("credential", "required-baseline", "control-plane"),
        RtcCapabilityCatalogEntry("provider.webhook", "required-baseline", "control-plane"),
        RtcCapabilityCatalogEntry("provider.event-normalization", "required-baseline", "control-plane"),
        RtcCapabilityCatalogEntry("health", "required-baseline", "control-plane"),
        RtcCapabilityCatalogEntry("media.audio", "required-baseline", "runtime-bridge"),
        RtcCapabilityCatalogEntry("media.video", "required-baseline", "runtime-bridge"),
        RtcCapabilityCatalogEntry("live.broadcast", "required-baseline", "cross-surface"),
        RtcCapabilityCatalogEntry("live.audience", "required-baseline", "cross-surface"),
        RtcCapabilityCatalogEntry("screen-share", "optional-advanced", "runtime-bridge"),
        RtcCapabilityCatalogEntry("recording", "optional-advanced", "control-plane"),
        RtcCapabilityCatalogEntry("artifact", "optional-advanced", "control-plane"),
        RtcCapabilityCatalogEntry("cloud-mix", "optional-advanced", "control-plane"),
        RtcCapabilityCatalogEntry("cdn-relay", "optional-advanced", "control-plane"),
        RtcCapabilityCatalogEntry("data-channel", "optional-advanced", "runtime-bridge"),
        RtcCapabilityCatalogEntry("transcription", "optional-advanced", "control-plane"),
        RtcCapabilityCatalogEntry("beauty", "optional-advanced", "runtime-bridge"),
        RtcCapabilityCatalogEntry("spatial-audio", "optional-advanced", "runtime-bridge"),
        RtcCapabilityCatalogEntry("e2ee", "optional-advanced", "runtime-bridge"),
        RtcCapabilityCatalogEntry("provider.active-query", "optional-advanced", "control-plane"),
    ]


def get_rtc_capability_catalog() -> list[RtcCapabilityCatalogEntry]:
    return RtcCapabilityCatalog.entries


def get_rtc_capability_descriptor(capability_key: str) -> Optional[RtcCapabilityCatalogEntry]:
    for entry in RtcCapabilityCatalog.entries:
        if entry.capabilityKey == capability_key:
            return entry

    return None
